app/assets/javascripts/discourse/app/controllers/preferences/profile.js

```diff
@@ -53,6 +60,11 @@ export default Controller.extend({
     });
   },
 
+  @discourseComputed("currentUser")
+  showEnforcedRequiredFieldsNotice(user) {
```

Comment:
Ah. Good point. Agree. Probably confusing to just have the "saved" flash there in this case.

Reply:
This is still happening after your changes. I think we need to update _something_ on the user model to let them navigate away from the page, I am guessing `user.needs_required_fields_check` since that's used in willTransition on the route.